Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm

```diff
@@ -193,23 +323,34 @@
         localizedDescription = WEB_UI_STRING("An unsupported `manifest_version` was specified.", "WKWebExtensionErrorUnsupportedManifestVersion description");
         break;
 
+    case Error::InvalidAction:
+        errorCode = _WKWebExtensionErrorInvalidManifestEntry;
+        if (usesManifestVersion(3))
+            localizedDescription = WEB_UI_STRING("Missing or empty `action` manifest entry.", "WKWebExtensionErrorInvalidManifestEntry description for action only");
+        else
+            localizedDescription = WEB_UI_STRING("Missing or empty `browser_action` or `page_action` manifest entry.", "WKWebExtensionErrorInvalidManifestEntry description for browser_action or page_action");
+        break;
+
     case Error::InvalidActionIcon:
-        errorCode = _WKWebExtensionErrorInvalidActionIcon;
-        localizedDescription = WEB_UI_STRING("Empty or invalid `default_icon` for the `action`, `browser_action`, or `page_action` manifest entry.", "WKWebExtensionErrorInvalidActionIcon description");
+        errorCode = _WKWebExtensionErrorInvalidManifestEntry;
+        if (usesManifestVersion(3))
+            localizedDescription = WEB_UI_STRING("Empty or invalid `default_icon` for the `action` manifest entry.", "WKWebExtensionErrorInvalidManifestEntry description for default_icon in action only");
+        else
+            localizedDescription = WEB_UI_STRING("Empty or invalid `default_icon` for the `browser_action` or `page_action` manifest entry.", "WKWebExtensionErrorInvalidManifestEntry description for default_icon in browser_action or page_action");
         break;
 
     case Error::InvalidBackgroundContent:
-        errorCode = _WKWebExtensionErrorInvalidBackgroundContent;
-        localizedDescription = WEB_UI_STRING("Empty or invalid `background` manifest entry.", "WKWebExtensionErrorInvalidBackgroundContent description");
+        errorCode = _WKWebExtensionErrorInvalidManifestEntry;
```

Comment:
Ah I see - you replaced all of the individual error codes with ...InvalidManifestEntry.

Reply:
Yep